jerry-core/jcontext/jcontext.h

```diff
@@ -116,6 +117,84 @@ typedef struct
 } jerry_context_t;
 
 /**
+ * Description of jerry instance, which contains the meta data and buffer.
+ */
+struct jerry_instance_t
+{
+  struct
+  {
+    uint32_t heap_size; /**< size of the heap */
+    uint8_t *context_p; /**< point to the entrance of the context in buffer */
+#ifndef JERRY_SYSTEM_ALLOCATOR
+    uint8_t *heap_p; /**< point to the entrance of the heap in buffer */
+#endif /* !JERRY_SYSTEM_ALLOCATOR */
+#ifndef CONFIG_ECMA_LCACHE_DISABLE
+    uint8_t *lcache_p; /**< point to the entrance of the lcache in buffer */
+#endif /* !CONFIG_ECMA_LCACHE_DISABLE */
+  } head;
+  uint8_t buffer[]; /**< the flexible array for storing context, heap and lcache */
```

Comment:
Hm an idea: if context_p is the first member of buffer, it could be accessed without context_p. That would be one less memory read.

    in jerry_context_get_current_context():
    return (jerry_context_t *) (jerry_port_get_current_instance () + 1);

Is buffer[] supported by C99? And do we need this member at all (`jerry_port_get_current_instance () + 1` is the start for the data)? This way we would not need the "head" nested structure as well.

Furthermore inlining this could be good for optimization, since the compiler is clever enough to change `jerry_context_get_current_context()->something` to a `jerry_port_get_current_instance ()+offset` read (no size increase, better performance).

Another alternative: we could make heap_size, heap_p, lcache_p part of the jerry_context_t.